ec2-ip

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+set -eu
+
+BUCKET=ec2-cache.binaris.com
+DEFAULT_FZF_QUERY=${1:-${realm:-$USER}}
+
+# Requirements:
+# - fzf
+# - aws cli
+
+function get_cache() {
+    # Set you bucket name here
+    aws s3 cp --quiet s3://$BUCKET/instances.fzf >(cat)
+}
+
+function choose() {
+    cache="$(get_cache)"
+    chosen=$(echo -e "$cache\n%refresh ec2 cache" | fzf -q "$DEFAULT_FZF_QUERY")
+}
+
+while true; do
```

Comment:
Logic might be simpler if the loop condition is based on `$chosen`, e.g. something like (untested, but what could possibly go wrong writing a shell script like this?):
```suggestion
until
    choose
    [[ "$chosen" != "%refresh ec2 cache" ]];  
do
    echo "Refreshing EC2 fuzzy cache..."
    aws lambda invoke --region ${region:-eu-central-1} --function-name ec2-fzf-${region:-eu-central-1}-discover_instances /dev/null >/dev/null
done
if [[ -z "$chosen" ]]; then exit 1;
```

Reply:
Only one `fi` was missing 👍